server/coordinator/factory.go

```diff
@@ -126,22 +126,32 @@ func (f *Factory) makeCreateTableProcedure(ctx context.Context, request CreateTa
 	}
 	snapshot := request.ClusterMetadata.GetClusterSnapshot()
 
-	shards, err := f.shardPicker.PickShards(ctx, snapshot, 1)
+	var targetShardID storage.ShardID
+	shardID, exists, err := request.ClusterMetadata.GetAssignTable(ctx, request.SourceReq.SchemaName, request.SourceReq.Name)
 	if err != nil {
-		f.logger.Error("pick table shard", zap.Error(err))
-		return nil, errors.WithMessage(err, "pick table shard")
+		return nil, err
 	}
-	if len(shards) != 1 {
-		f.logger.Error("pick table shards length not equal 1", zap.Int("shards", len(shards)))
-		return nil, errors.WithMessagef(procedure.ErrPickShard, "pick table shard, shards length:%d", len(shards))
+	if exists {
+		targetShardID = shardID
+	} else {
+		shards, err := f.shardPicker.PickShards(ctx, snapshot, 1)
+		if err != nil {
+			f.logger.Error("pick table shard", zap.Error(err))
+			return nil, errors.WithMessage(err, "pick table shard")
+		}
+		if len(shards) != 1 {
+			f.logger.Error("pick table shards length not equal 1", zap.Int("shards", len(shards)))
+			return nil, errors.WithMessagef(procedure.ErrPickShard, "pick table shard, shards length:%d", len(shards))
+		}
+		targetShardID = shards[0].ID
```

Comment:
I guess we can encapsulate the table assignment to the `ShardPicker`. Just provide a new `ShardPicker` wrapper around the original one, and do table assignment query/creation in the wrapper.